assets/agenda/components/AgendaPreviewEvent.tsx

```diff
@@ -24,110 +23,112 @@ class AgendaPreviewEventComponent extends React.Component<any, any> {
 
         this.state = {
             loading: true,
-            expanded: false,
+            expandedEvents: {},
         };
+
         this.toggleExpanded = this.toggleExpanded.bind(this);
+        this.reloadEvent = this.reloadEvent.bind(this);
     }
 
     componentDidMount() {
         this.reloadEvent();
     }
 
     componentDidUpdate(prevProps: any) {
-        if (get(prevProps.item, 'event_id') !== get(this.props.item, 'event_id')) {
+        if (get(prevProps.item, 'event_ids') !== get(this.props.item, 'event_ids')) {
             this.reloadEvent();
         }
     }
 
     reloadEvent() {
-        this.setState({loading: true}, () => {
-            this.props
-                .fetchEvent(this.props.item.event_id)
-                .finally(() => {
-                    this.setState({loading: false});
-                });
-        });
+        const {item, fetchEvent} = this.props;
+        const eventIds = item.event_ids || [];
+
+        this.setState({loading: true});
```

Comment:
Rather than sending a request for each Event, you could use the search endpoint to search by ids.
i.e.

```ts
// file: assets/agenda/actions.ts
export function fetchItemsById(ids: Array<string>): Promise<IRestApiResponse<IAgendaItem>> {
    return server.get(`/agenda/search?ids=${ids.join(',')}`);
}

export function fetchItemsByIdToRedux(ids: Array<string>) {
    return (dispatch: any) => {
        fetchItemsById(ids).then((response) => {
            dispatch(recieveItems(response));
        });
    };
}
```

And then use it here:
```ts
if (item.event_ids == null || item.event_ids.length == 0) {
    return;
}

this.setState({loading: true});
this.props
    .fetchItemsByIdToRedux(item.event_ids)
    .finally(() => {
        this.setState({loading: false});
    });
```

Reply:
For this endpoint, we need to update the backend code as well. afaik, we can't handle the `_ids` param in this endpoint.

Reply:
It _should_ work, if not it needs to be fixed.
https://github.com/superdesk/newsroom-core/blob/async/newsroom/search/types.py#L196
https://github.com/superdesk/newsroom-core/blob/async/newsroom/search/filters.py#L158

Reply:
it's, working, Thanks